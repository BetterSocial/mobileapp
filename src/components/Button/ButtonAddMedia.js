import * as React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

import {colors} from '../../utils/colors';

const ButtonAddMedia = ({onPress, style, label, labelStyle}) => {
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      <Text style={labelStyle}>{label}</Text>
    </TouchableOpacity>
  );
};

export default ButtonAddMedia;

const styles = StyleSheet.create({
  container: {
    paddingVertical: 32,
    alignItems: 'center',
    borderWidth: 2,
    borderColor: colors.alto,
    borderRadius: 8,
    borderStyle: 'dashed',
  },
});
