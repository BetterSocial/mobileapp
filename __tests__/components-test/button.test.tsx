import React from 'react';
import {shallow, configure} from 'enzyme';
import renderer from 'react-test-renderer';
import ButtonAddMedia from '../../src/components/Button/ButtonAddMedia';
import Adapter from 'enzyme-adapter-react-16';

import toJson from 'enzyme-to-json';
configure({adapter: new Adapter()});
describe('button', () => {
  it('button snapshot', () => {
    const component = shallow(
      <ButtonAddMedia
        onPress={() => {}}
        style={{}}
        label="+add more photos"
        labelStyle={{}}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Button renders correctly', () => {
    const tree = renderer
      .create(
        <ButtonAddMedia
          onPress={() => {}}
          style={{}}
          label="+add more photos"
          labelStyle={{}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
