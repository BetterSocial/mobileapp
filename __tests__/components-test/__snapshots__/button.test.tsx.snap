// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button Button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E0E0E0",
      "borderRadius": 8,
      "borderStyle": "dashed",
      "borderWidth": 2,
      "opacity": 1,
      "paddingVertical": 32,
    }
  }
>
  <Text
    style={Object {}}
  >
    +add more photos
  </Text>
</View>
`;

exports[`button button snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#E0E0E0",
        "borderRadius": 8,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "paddingVertical": 32,
      },
      Object {},
    ]
  }
>
  <Text
    style={Object {}}
  >
    +add more photos
  </Text>
</ForwardRef>
`;
